src/components/ReducedMotion.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from "react"
+import { useContext } from "react"
+import { MotionContext } from "../motion/context/MotionContext"
+
+interface Props {
+    children?: any
+
+    /**
+     * Can be used to explicitly set whether we're in reduced motion mode. Set
+     * as undefined to resume device detection.
+     */
+    enabled?: boolean | undefined
+}
+
+/**
+ * Define accessibility options for a tree. Can be used to force the tree into Reduced Motion mode,
+ * or disable device detection.
+ *
+ * @internal
+ */
+export function ReducedMotion({ children, enabled }: Props) {
+    let context = useContext(MotionContext)
+
+    context = {
```

Comment:
maybe worth memoizing this? at the moment each ReducedMotion rerender will scan the whole subtree to update context value, most likely this ReducedMotion should always be put on the top level of the app (so it actually wouldnt ever - or really rarely - be rerendered), but it's not a given

Reply:
Yeah great point thanks